rust/src/mime/smtp.rs

```diff
@@ -614,9 +616,30 @@ fn mime_smtp_parse_line(
                     || ctx.content_type == MimeSmtpContentType::Html
                     || ctx.content_type == MimeSmtpContentType::Message
                 {
-                    mime_smtp_find_url_strings(ctx, full);
+                    match ctx.encoding {
+                        MimeSmtpEncoding::Plain => {
+                            mime_smtp_find_url_strings(ctx, full);
+                        }
+                        MimeSmtpEncoding::QuotedPrintable => {
+                            mime_smtp_find_url_strings(ctx, full);
+                        }
+                        MimeSmtpEncoding::Base64 => {
+                            if unsafe { MIME_SMTP_CONFIG_DECODE_BASE64 } {
+                                if let Some(ref mut decoder) = &mut ctx.decoder {
```

Comment:
is this previously decoded? Or is this decoded in parallel to file decoding?

Reply:
We are here because `if ctx.filename.is_empty()`

So, there is no file decoding.

But we may want to factor some code together, is that what you mean ?